compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTupleMemberSymbol.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.impl.SymbolFactory;
+import io.ballerina.compiler.api.symbols.AnnotationAttachmentSymbol;
+import io.ballerina.compiler.api.symbols.AnnotationSymbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TupleMemberTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BAnnotationAttachmentSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BVarSymbol;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a tuple member type symbol.
+ *
+ * @since 2201.4.0
+ */
+public class BallerinaTupleMemberSymbol extends BallerinaSymbol implements TupleMemberTypeSymbol {
+
+    private List<AnnotationSymbol> annots;
+    private List<AnnotationAttachmentSymbol> annotAttachments;
+    private final TypeSymbol type;
+
+    public BallerinaTupleMemberSymbol(CompilerContext context,
+                                      String name,
+                                      BVarSymbol symbol,
+                                      TypeSymbol type) {
+        super(name, SymbolKind.TUPLE_MEMBER_TYPE, symbol, context);
+        this.type = type;
+    }
+
+    public TypeSymbol getType() {
+        return this.type;
+    }
+
+    @Override
+    public List<AnnotationSymbol> annotations() {
```

Comment:
Not directly related, but haven't we deprecated this?

Reply:
Yeah, need to deprecate this

Reply:
Do we have an issue for this?

Reply:
Just created a one. https://github.com/ballerina-platform/ballerina-lang/issues/39181